library/ssl_tls13_client.c

```diff
@@ -963,6 +968,32 @@ int mbedtls_ssl_tls13_write_identities_of_pre_shared_key_ext(
 
         p += output_len;
         l_binders_len += 1 + PSA_HASH_LENGTH( hash_alg );
+
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+        MBEDTLS_SSL_DEBUG_MSG(
+            1, ( "Set hs psk for early data when writing the first psk" ) );
+
+        ret = ssl_tls13_ticket_get_psk( ssl, &hash_alg, &psk, &psk_len );
+        if( ret != 0 )
+        {
+            MBEDTLS_SSL_DEBUG_RET(
+                    1, "ssl_tls13_ticket_get_psk", ret );
+            return ( ret );
+        }
+
+        ret = mbedtls_ssl_set_hs_psk( ssl, psk, psk_len );
+        if( ret  != 0 )
+        {
+            MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_set_hs_psk", ret );
+            return ( ret );
+        }
+
+        ciphersuite_info = mbedtls_ssl_ciphersuite_from_id(
+                                ssl->session_negotiate->ciphersuite );
+        ssl->handshake->ciphersuite_info = ciphersuite_info;
+        ssl->handshake->key_exchange_mode =
+                    MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ALL;
```

Comment:
```suggestion
```

It is not needed

Reply:
We need this mode be set, cause we need the psk enabled check in early data processing, which is earlier than the process_server_hello().

Reply:
That's `conf->kex_mode` not `handshake->key_exchange_mode` .  

`handshake->key_exchange_mode` was determined by negotiate. The value is only valid after server hello. 

And what's  breaked without it?

Reply:
> That's `conf->kex_mode` not `handshake->key_exchange_mode` .
> 
---------No, it need handshake->key_exchange_mode, not only conf->kex_mode.
> `handshake->key_exchange_mode` was determined by negotiate. The value is only valid after server hello.
> 
> And what's breaked without it?
-------The basic good test case, client side will report error: "! bad client exit code (expected 0, got 1)", server side report: "Error in handshake: Decryption has failed."

Reply:
So, add comments to describe the issue.